app/components/form/fields/NumberField.tsx

```diff
@@ -80,7 +80,18 @@ export const NumberFieldInner = <
   const {
     field,
     fieldState: { error },
-  } = useController({ name, control, rules: { required, validate } })
+  } = useController({
+    name,
+    control,
+    rules: {
+      required,
+      // it seems we need special logic to enforce required on NaN
+      validate(value, values) {
+        if (required && Number.isNaN(value)) return `${label} is required`
+        return validate?.(value, values)
+      },
+    },
+  })
```

Comment:
I think `NaN` wasn't failing RHF's check for a required field the way you'd hope